docs/source/_src_user_guide/state_propagation/environment_setup/thrust_refactor/thrust_refactor.rst

```diff
@@ -0,0 +1,168 @@
+
+Converting thrust code
+----------------------
+
+In the old code, you typically defined a thrust acceleration through:
+
+    .. tabs::
+
+         .. tab:: Python
+
+          .. literalinclude:: /_src_snippets/simulation/environment_setup/thrust_refactor/thrust_acceleration_old.py          
+             :language: python
+
+
+In the new code, the equivalent functionality is provided through:
+
+    .. tabs::
+
+         .. tab:: Python
+
+          .. literalinclude:: /_src_snippets/simulation/environment_setup/thrust_refactor/thrust_acceleration_new.py          
+             :language: python
+
+Here, we have chosen to define the rotation model settings, and create the rotation model, after creating the ``SystemOfBodies``, as discussed  :ref:`here <decorate_empty_body>`.
+
+Details on the functioning of the two pieces of code can be found on the dedicated thrust pages of the new code, and the legacy code (see above). The steps to take:
+
+* Convert your thrust direction settings to rotation model settings. See below for a one-to-one correspondence betwee individual models
+* Using the same ``thrust_magnitude_settings`` as before, create an engine model using the :func:`~tudatpy.numerical_simulation.environment_setup.add_engine_model` function. In addition to containing information on the thrust magnitude, the engine object holds information on:
+
+  * The name of the engine (used to link it to the acceleration, see below)
+  * The body-fixed thrust direction (for a time-variable body-fixed thrust direction, use the :func:`~tudatpy.numerical_simulation.environment_setup.add_engine_model` instead.)
+* Create the thrust acceleration, by specifying which engine(s) should be used in the calculation of the thrust acceleration. Here, the name provided to the engine model is used.
+
+Converting thrust direction settings
+------------------------------------
+
+Each of the old thrust direction settings (in the ``propagation_setup.thrust`` submodule) is now reformulated as a rotation model setting. In addition to the inputs from the corresponding thrust direction settings, the rotation model settings require the ``base_frame`` (typically J2000 or ECLIPJ2000), and the ``target_frame`` to be defined. The ``target_frame`` is simply the identifier that the code uses for the body-fixed frame of the vehicle, and may be selected at will by the user.
+
+**propagation_setup.thrust.thrust_from_existing_body_orientation** This option is now moot, *by definition* all thrust models now use the existing body orientation. If you were using this option, now additional action is needed.
+
+**propagation_setup.thrust.custom_thrust_orientation**  This option is now replaced with the equivalent :func:`~tudatpy.numerical_simulation.environment_setup.rotation_model.custom_rotation_model` option, with the addition of the frame identiers specifief above. It defines the body-fixed orientation in a fully user-defined manner.
```

Comment:
...addition of the frame ~~identiers specifief~~ *identifiers specified* above...
(The same error was copy-pasted in the following two paragraphs as well)